---
title: ActionDispatch::ExceptionWrapper
layout: default
---
<div class="main">
    <div class="banner">
        
            <span>Ruby on Rails 8.0.0</span><br />
        
        <div class="type">Class</div>
        <h1>
            ActionDispatch::ExceptionWrapper
            
                <span class="parent">&lt;
                    
                    <a href="../Object.html">Object</a>
                    
                </span>
            
        </h1>
        <ul class="files">
            
            <li><a href="../../files/actionpack/lib/action_dispatch/middleware/exception_wrapper_rb.html">actionpack/lib/action_dispatch/middleware/exception_wrapper.rb</a></li>
            
        </ul>
    </div>
    <div id="bodyContent">
        <div id="content">
  

  

  
  


  

  
    <h2 id="methods">Methods</h2>
    <ul>
      
        <li>
          <a href="#method-i-actions">actions</a>
        </li>
      
        <li>
          <a href="#method-i-annotated_source_code">annotated_source_code</a>
        </li>
      
        <li>
          <a href="#method-i-application_trace">application_trace</a>
        </li>
      
        <li>
          <a href="#method-i-corrections">corrections</a>
        </li>
      
        <li>
          <a href="#method-i-exception_id">exception_id</a>
        </li>
      
        <li>
          <a href="#method-i-exception_inspect">exception_inspect</a>
        </li>
      
        <li>
          <a href="#method-i-exception_name">exception_name</a>
        </li>
      
        <li>
          <a href="#method-i-exception_trace">exception_trace</a>
        </li>
      
        <li>
          <a href="#method-i-failures">failures</a>
        </li>
      
        <li>
          <a href="#method-i-file_name">file_name</a>
        </li>
      
        <li>
          <a href="#method-i-framework_trace">framework_trace</a>
        </li>
      
        <li>
          <a href="#method-i-full_trace">full_trace</a>
        </li>
      
        <li>
          <a href="#method-i-has_cause-3F">has_cause?</a>
        </li>
      
        <li>
          <a href="#method-i-has_corrections-3F">has_corrections?</a>
        </li>
      
        <li>
          <a href="#method-i-line_number">line_number</a>
        </li>
      
        <li>
          <a href="#method-i-message">message</a>
        </li>
      
        <li>
          <a href="#method-c-new">new</a>
        </li>
      
        <li>
          <a href="#method-i-original_message">original_message</a>
        </li>
      
        <li>
          <a href="#method-i-rescue_response-3F">rescue_response?</a>
        </li>
      
        <li>
          <a href="#method-i-rescue_template">rescue_template</a>
        </li>
      
        <li>
          <a href="#method-i-routing_error-3F">routing_error?</a>
        </li>
      
        <li>
          <a href="#method-i-show-3F">show?</a>
        </li>
      
        <li>
          <a href="#method-i-source_extracts">source_extracts</a>
        </li>
      
        <li>
          <a href="#method-i-source_to_show_id">source_to_show_id</a>
        </li>
      
        <li>
          <a href="#method-i-status_code">status_code</a>
        </li>
      
        <li>
          <a href="#method-c-status_code_for_exception">status_code_for_exception</a>
        </li>
      
        <li>
          <a href="#method-i-sub_template_message">sub_template_message</a>
        </li>
      
        <li>
          <a href="#method-i-template_error-3F">template_error?</a>
        </li>
      
        <li>
          <a href="#method-i-trace_to_show">trace_to_show</a>
        </li>
      
        <li>
          <a href="#method-i-traces">traces</a>
        </li>
      
        <li>
          <a href="#method-i-unwrapped_exception">unwrapped_exception</a>
        </li>
      
    </ul>
  

  

  
    

    

    

    
      <!-- Section attributes -->
      <h2 id="attributes">Attributes</h2>
      <table border='0' cellpadding='5'>
        
          <tr valign='top'>
            <td class='attr-rw'>
              [R]
            </td>
            <td class='attr-name'>backtrace_cleaner</td>
            <td class='attr-desc'></td>
          </tr>
        
          <tr valign='top'>
            <td class='attr-rw'>
              [R]
            </td>
            <td class='attr-name'>exception</td>
            <td class='attr-desc'></td>
          </tr>
        
          <tr valign='top'>
            <td class='attr-rw'>
              [R]
            </td>
            <td class='attr-name'>exception_class_name</td>
            <td class='attr-desc'></td>
          </tr>
        
          <tr valign='top'>
            <td class='attr-rw'>
              [R]
            </td>
            <td class='attr-name'>wrapped_causes</td>
            <td class='attr-desc'></td>
          </tr>
        
      </table>
    

    <!-- Methods -->
    
      <h2 id="class-public-methods">Class Public methods</h2>
      
        <div class="method">
          <h3 id="method-c-new">
            
              new(backtrace_cleaner, exception)
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 50
    def initialize(backtrace_cleaner, exception)
      @backtrace_cleaner = backtrace_cleaner
      @exception_class_name = exception.class.name
      @wrapped_causes = wrapped_causes_for(exception, backtrace_cleaner)
      @exception = exception
      if exception.is_a?(SyntaxError)
        @exception = ActiveSupport::SyntaxErrorProxy.new(exception)
      end
      @backtrace = build_backtrace
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L50" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-c-status_code_for_exception">
            
              status_code_for_exception(class_name)
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 175
    def self.status_code_for_exception(class_name)
      Rack::Utils.status_code(@@rescue_responses[class_name])
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L175" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
      
    
      <h2 id="instance-public-methods">Instance Public methods</h2>
      
        <div class="method">
          <h3 id="method-i-actions">
            
              actions()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 101
    def actions
      ActiveSupport::ActionableError.actions(@exception)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L101" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-annotated_source_code">
            
              annotated_source_code()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 113
    def annotated_source_code
      if exception.respond_to?(:annotated_source_code)
        exception.annotated_source_code
      else
        []
      end
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L113" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-application_trace">
            
              application_trace()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 135
    def application_trace
      clean_backtrace(:silent)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L135" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-corrections">
            
              corrections()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 89
    def corrections
      @exception.corrections
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L89" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-exception_id">
            
              exception_id()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 228
    def exception_id
      exception.object_id
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L228" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-exception_inspect">
            
              exception_inspect()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 224
    def exception_inspect
      exception.inspect
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L224" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-exception_name">
            
              exception_name()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 216
    def exception_name
      exception.cause.class.to_s
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L216" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-exception_trace">
            
              exception_trace()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 129
    def exception_trace
      trace = application_trace
      trace = framework_trace if trace.empty? &amp;&amp; !silent_exceptions.include?(@exception_class_name)
      trace
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L129" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-failures">
            
              failures()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 77
    def failures
      @exception.failures
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L77" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-file_name">
            
              file_name()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 93
    def file_name
      @exception.file_name
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L93" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-framework_trace">
            
              framework_trace()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 139
    def framework_trace
      clean_backtrace(:noise)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L139" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-full_trace">
            
              full_trace()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 143
    def full_trace
      clean_backtrace(:all)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L143" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-has_cause-3F">
            
              has_cause?()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 73
    def has_cause?
      @exception.cause
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L73" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-has_corrections-3F">
            
              has_corrections?()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 81
    def has_corrections?
      @exception.respond_to?(:original_message) &amp;&amp; @exception.respond_to?(:corrections)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L81" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-line_number">
            
              line_number()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 97
    def line_number
      @exception.line_number
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L97" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-message">
            
              message()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 220
    def message
      exception.message
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L220" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-original_message">
            
              original_message()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 85
    def original_message
      @exception.original_message
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L85" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-rescue_response-3F">
            
              rescue_response?()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 194
    def rescue_response?
      @@rescue_responses.key?(exception.class.name)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L194" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-rescue_template">
            
              rescue_template()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 121
    def rescue_template
      @@rescue_templates[@exception_class_name]
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L121" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-routing_error-3F">
            
              routing_error?()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 61
    def routing_error?
      @exception.is_a?(ActionController::RoutingError)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L61" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-show-3F">
            
              show?(request)
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 179
    def show?(request)
      # We&#39;re treating `nil` as &quot;unset&quot;, and we want the default setting to be `:all`.
      # This logic should be extracted to `env_config` and calculated once.
      config = request.get_header(&quot;action_dispatch.show_exceptions&quot;)

      case config
      when :none
        false
      when :rescuable
        rescue_response?
      else
        true
      end
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L179" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-source_extracts">
            
              source_extracts()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 198
    def source_extracts
      backtrace.map do |trace|
        extract_source(trace)
      end
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L198" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-source_to_show_id">
            
              source_to_show_id()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 212
    def source_to_show_id
      (traces[trace_to_show].first || {})[:id]
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L212" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-status_code">
            
              status_code()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 125
    def status_code
      self.class.status_code_for_exception(unwrapped_exception.class.name)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L125" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-sub_template_message">
            
              sub_template_message()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 69
    def sub_template_message
      @exception.sub_template_message
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L69" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-template_error-3F">
            
              template_error?()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 65
    def template_error?
      @exception.is_a?(ActionView::Template::Error)
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L65" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-trace_to_show">
            
              trace_to_show()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 204
    def trace_to_show
      if traces[&quot;Application Trace&quot;].empty? &amp;&amp; rescue_template != &quot;routing_error&quot;
        &quot;Full Trace&quot;
      else
        &quot;Application Trace&quot;
      end
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L204" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-traces">
            
              traces()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 147
    def traces
      application_trace_with_ids = []
      framework_trace_with_ids = []
      full_trace_with_ids = []

      full_trace.each_with_index do |trace, idx|
        trace_with_id = {
          exception_object_id: @exception.object_id,
          id: idx,
          trace: trace
        }

        if application_trace.include?(trace)
          application_trace_with_ids &lt;&lt; trace_with_id
        else
          framework_trace_with_ids &lt;&lt; trace_with_id
        end

        full_trace_with_ids &lt;&lt; trace_with_id
      end

      {
        &quot;Application Trace&quot; =&gt; application_trace_with_ids,
        &quot;Framework Trace&quot; =&gt; framework_trace_with_ids,
        &quot;Full Trace&quot; =&gt; full_trace_with_ids
      }
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L147" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
        <div class="method">
          <h3 id="method-i-unwrapped_exception">
            
              unwrapped_exception()
            
          </h3>

          
            <div class="description">
              
            </div>
          

          

          

          
          
            <details class="method__source">
              <summary>
                <span class="label">üìù Source code</span>
              </summary>

              <pre><code class="ruby"># File actionpack/lib/action_dispatch/middleware/exception_wrapper.rb, line 105
    def unwrapped_exception
      if wrapper_exceptions.include?(@exception_class_name)
        @exception.cause
      else
        @exception
      end
    end</code></pre>
              
                <a href="https://github.com/rails/rails/blob/dd8f7185faeca6ee968a6e9367f6d8601a83b8db/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L105" target="_blank" class="github_url">üîé See on GitHub</a>
              
            </details>
          
        </div>
        
      
    
  
</div>

    </div>
</div>
